src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -36,35 +42,39 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
       .endpoint(getEndpoint)
       .buildClient()
 
+  import spark.implicits._
+  lazy val df: DataFrame = Seq(
+    (Seq(""),Seq("")),
+  ).toDF("lang", "text")
+
   protected def transformTextRows(toRow: TAResponseV4[T] => Row)
                                  (rows: Iterator[Row]): Iterator[Row] = {
+
     rows.map { row =>
       val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
       Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
     }}
 
   override def transform(dataset: Dataset[_]): DataFrame = {
+    val batchedDF = new FixedMiniBatchTransformer().setBatchSize(10).transform(dataset.coalesce(1))

```

Comment:
This number 10 --> pass in as a configuration number && set up a default number as 5 or 10.